modules/data.atmosphere/R/met_temporal_downscale.Gaussian_ensemble.R

```diff
@@ -0,0 +1,398 @@
+        daily.swdn[x] <- mean(swsource[(x * daily_step - daily_step + 1):(x * daily_step)])
+      }
+      
+      # creating the sine wave
+      for (i in seq_along(daily.swdn)) {
+        t <- seq(from = pi/day_len[i], to = pi, by = pi/day_len[i])
+        wav <- ((daily.swdn[i] * (24/day_len[i]))/0.637) * sin(t)
+        
+        # swdn = 0 without sunlight
+        srs <- eph$sunrise
+        hr <- substr(srs[i], 1, 2)
+        hr <- as.numeric(hr)
+        hr <- hr + utc_diff
+        
+        l <- vector()
+        for (n in seq_along(1:hr)) {
+          l[n] <- 0
+        }
+        for (n in seq_along(wav)) {
+          l[n + hr] <- wav[n]
+        }
+        for (n in seq_along(1:(24 - (length(wav) + hr)))) {
+          l[n + hr + length(wav)] <- 0
+        }
+        
+        swdn <- append(swdn, l)
+      }
+      
+      swflux <- vector()
+      sw_step <- length(swdn)/reso_len
+      for (x in 1:reso_len) {
+        swflux[x] <- mean(swdn[(x * sw_step - sw_step + 1):(x * sw_step)])
+      }
+      swflux[swflux < 0] <- 0
+    }
+    
+    # The spline swdn_method uses spline interpolation to connect existing values and
+    # downscale
+    if (swdn_method == "spline") {
+      tem.met <- vector()
+      for (x in seq(from = 0, to = nrow(train), by = div)) {
+        tem.met[x] <- swsource[x/div]
+      }
+      
+      swflux <- vector()
+      swflux <- zoo::na.spline(tem.met)
+      swflux[swflux < 0] <- 0
+    }
+    
+    # The Waichler swdn_method doesn't need averaged SW train values, it sources SW
+    # downwelling flux based on Tmax-Tmin and Precipitation Reference is Waichler and
+    # Wigtosa 2003. Our no-precip coefficient is 2 instead of 1 (fits the data
+    # better)
+    if (swdn_method == "Waichler") {
+      inter <- paste0(reso, " hour")
+      days <- seq(as.POSIXct(paste0(eph_year, "-01-01 00:00:00", tz = "US/Central")), 
+                  as.POSIXct(paste0(eph_year, "-12-31 18:00:00", tz = "US/Central")), 
+                  by = inter)
+      
+      Z <- RAtmosphere::SZA(days, lat_train, lon_train)
+      I <- 1000 * aspace::cos_d(Z)
+      m <- vector()
+      for (i in seq_along(1:12)) {
+        m[i] <- Hmisc::monthDays(as.Date(paste0(year, "-", i, "-01")))
+      }
+      bmlist <- vector()
+      
+      Bm <- c(0.2089, 0.2857, 0.2689, 0.2137, 0.1925, 0.2209, 0.2527, 0.2495, 
+              0.2232, 0.1728, 0.1424, 0.1422)
+      for (x in seq_along(Bm)) {
+        mlen <- list()
+        mlen <- rep(Bm[x], m[x] * 24/reso)
+        bmlist <- append(bmlist, mlen)
+      }
+      A <- 0.73
+      C <- 0.7
+      hdry <- vector()
+      for (i in seq_along(precip)) {
+        if (precip[i] > 0) {
+          p <- 0.65
+        }
+        if (precip[i] == 0) {
+          p <- 2
+        }
+        hdry[i] <- A * p * (1 - exp(-1 * bmlist[i] * ((temp_max[i] - temp_min[i])^C)))
+      }
+      hdry[hdry < 0] <- 0
+      swflux <- hdry * I
+      swflux[swflux < 0] <- 0
+    }
+    
```

Comment:
Not with the Waichler method, which uses ensemble specific temp_max, temp_min, and precip